.github/workflows/docker_build.yml

```diff
@@ -0,0 +1,53 @@
+# This workflow will build a docker image for this repository and publish it to the ghcr
+# We also implement Docker Layer Caching using GHA(GitHub Actions cache)
+# For more information, see: https://www.kenmuse.com/blog/implementing-docker-layer-caching-in-github-actions/
+name: Docker Build
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ "develop" ]
+    tags: [ 'v*.*.*' ]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-container:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name == 'push'
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
```

Comment:
Could you add the following under `images:`
```
tags: |
    type=semver,pattern={{version}}
    type=semver,pattern={{major}}.{{minor}} 
    type=semver,pattern={{major}}
    type=ref,event=branch
    ref=main
    ref=develop
    type=sha
    latest
```
This will make sure we have docker image for the following tags `develop`, `main`, `latest`, and version of the tag I don't know if we need to have docker images created and tagged per feature branch as well.

Reply:
Based on our discussion on slack, we would like to add docker build step in `ci` workflow, create a docker image for develop, tag, and feature branch, and to use Registry cache since gha has a 10GB per repo storage limit.